packages/common/test/util.test.js

```diff
@@ -1,5 +1,365 @@
 import { expect } from 'chai';
-import { expandReferences, normalizeOauthConfig } from '../src/util';
+import {
+  enableMockClient,
+  expandReferences,
+  normalizeOauthConfig,
+  get,
+  post,
+  del,
+} from '../src/util';
+
+const client = enableMockClient('https://www.example.com');
+describe('request', () => {
+  it('should not set header content-type to application/json if body is string', async () => {
+    let request;
+    client
+      .intercept({
+        path: '/api',
+        method: 'POST',
+      })
+      .reply(200, r => {
+        request = r;
+        return '';
+      });
+
+    await post('/api', 'some string', {
+      baseUrl: 'https://www.example.com',
+    });
+
+    expect(request.headers).to.eql({});
+  });
+  it('should use baseUrl from options', async () => {
+    let request;
+    client
+      .intercept({
+        path: '/api',
+        method: 'GET',
+      })
+      .reply(200, r => {
+        request = r;
+        return {};
+      });
+
+    const response = await get('/api', { baseUrl: 'https://www.example.com' });
+
+    expect(request.path).to.eql('/api');
+    expect(response.code).to.eql(200);
+  });
+
+  it('should use fullUrlOrPath', async () => {
```

Comment:
Not a great test name - maybe `should accept an absolute url`. I would like the name to reference the previous test, because the two go hand-in-hand, so maybe both test names need thinking about together